src/frontend/components/view--goal-search.tsx

```diff
@@ -38,81 +38,83 @@ export default function GoalsSearchView({ filters, goals, total, description }:
     setActiveTopics(currentFilters)
   }
 
-  function filterGoals(resetOffset = false) {
```

Comment:
The useEffect hooks needed the function passed in as a dependency but in React passing in the function will cause the component to keep re-rendering. The useCallback hook memoizes the function so it won't cause the rerender. https://react.dev/reference/react/useCallback